recipes/lightning/meta.yaml

```diff
@@ -0,0 +1,82 @@
+{% set name = "lightning" %}
+{% set version = "2.0.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/lightning-{{ version }}.tar.gz
+  sha256: dfe158aa91ac139d8bdfccc7cdb627072e0052076ae9c0459c8fa12a028dbe6c
+
+build:
+  entry_points:
+    - lightning = lightning.app.cli.lightning_cli:main
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.8
+    - setuptools
+    - wheel
+  run:
+    - arrow <3.0,>=1.2.0
+    - beautifulsoup4 <6.0,>=4.8.0
+    - click <10.0
+    - croniter <1.4.0,>=1.3.0
+    - dateutils <2.0
+    - deepdiff <8.0,>=5.7.0
+    - fastapi <0.89.0
+    - fsspec <2024.0,>=2022.5.0
+    - inquirer <5.0,>=2.10.0
+    - jinja2 <5.0
+    - lightning-cloud >=0.5.31
+    - lightning-utilities <2.0,>=0.7.0
+    - numpy <3.0,>=1.17.2
+    - packaging
+    - psutil <7.0
+    - pydantic <3.0
+    - python >=3.8
+    - python-multipart
+    - pytorch <4.0,>=1.11.0
+    - pytorch-lightning
+    - pyyaml <8.0
+    - requests <4.0
+    - rich <15.0,>=12.3.0
+    - starlette <2.0
+    - starsessions <2.0,>=1.2.1
+    - torchmetrics <2.0,>=0.7.0
+    - tqdm <6.0,>=4.57.0
+    - traitlets <7.0,>=5.3.0
+    - typing-extensions <6.0,>=4.0.0
+    - urllib3 <3.0
+    - uvicorn <2.0
+    - websocket-client <3.0
+    - websockets <12.0
+
+test:
+  imports:
+    - lightning
+  commands:
+    - pip check
+    - lightning --help
```

Comment:
> This is timing out on the build, was this a problem previously?

Ok, after building and installing `lightning==2.0.0` locally and running `lightning --help`, there is a prompt that requires user input:

```
A newer version of lightning is available (2.0.1). Would you like to upgrade? [Y/n]:
```

Need to find a way to disable this prompt on the CLI, otherwise it'll block the build from completing.

Reply:
Using `LIGHTING_TESTING=1 lightning --help` in 466130035c9bccf610e68c9383b1b03878cc7a15 worked. @dopplershift, should be ok to merge now!